server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -4572,4 +4577,67 @@ public void testStressUpdateSameDocWhileGettingIt() throws IOException, Interrup
             }
         }
     }
+
+    public void testPruneOnlyDeletesAtMostLocalCheckpoint() throws Exception {
+        IOUtils.close(engine, store);
```

Comment:
can we try not to mess up with the class's engine ? I rather have a local one enclosed in a try with resources

Reply:
Done